inspirehep/utils/record_getter.py

```diff
@@ -105,3 +106,19 @@ def get_db_record(pid_type, recid):
     from inspirehep.modules.records.api import InspireRecord
     pid = PersistentIdentifier.get(pid_type, recid)
     return InspireRecord.get_record(pid.object_uuid)
+
+
+def get_db_records(pid_type, recids):
+    """Get an iterator on record metadata from the DB."""
+    recids = [str(recid) for recid in recids]
+
+    if not recids:
+        return iter([])
+
+    query = RecordMetadata.query.join(
+        PersistentIdentifier, RecordMetadata.id == PersistentIdentifier.object_uuid
+    ).filter(
+        PersistentIdentifier.pid_value.in_(recids), PersistentIdentifier.pid_type == pid_type
+    )
+
+    return (record.json for record in query.yield_per(100))
```

Comment:
Why not returning record directly (RecordMetadata instance)? That way would be more reusable

Reply:
again, there are already functions `get_db_record` and `get_es_records`. The goal was to act in the same way as those.